src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java

```diff
@@ -219,4 +218,42 @@ protected PasswordAuthentication getPasswordAuthentication() {
                 origin
         ).init();
     }
+
+    private static void check(Config config) throws ConfigException {
+        if (config == null || config.moneytransfer == null || config.database == null || config.deployment == null ||
+                config.platforms == null || config.database.writing == null || config.database.readonly == null ||
+                config.database.dialect == null || config.database.url == null || config.database.readonly.user == null ||
+                config.database.readonly.password == null || config.database.writing.user == null ||
+                config.database.writing.password == null || config.deployment.origin == null ||
+                config.moneytransfer.notificationMailAddress == null || config.moneytransfer.parsingPassword == null) {
+            throw new ConfigException("Configuration file incomplete");
+        }
+
+        if (config.database.maintainInterval == 0) {
+            config.database.maintainInterval = 24;
+        } else if (config.database.maintainInterval < 0) {
+            throw new ConfigException("negative maintainInterval of database is not valid");
+        }
+
+
+        if (config.moneytransfer.scheduleInterval == 0) {
+            config.moneytransfer.scheduleInterval = 7;
+        } else if (config.moneytransfer.scheduleInterval < 0) {
+            throw new ConfigException("A negative schedule interval is invalid.");
+        }
+
+        String codePatternStr = ".+@.+[.].+";
+        Pattern codePattern = Pattern.compile(codePatternStr);
+        Matcher codeMatcher = codePattern.matcher(config.moneytransfer.notificationMailAddress);
+        if (!codeMatcher.find()) {
+            throw new ConfigException("Notification Mail Address is invalid.");
+        }
+
+        for (ConfigPlatform platform : config.platforms) {
+            if (platform.type == null) {
```

Comment:
this can be improved, platform.type just says if this is a dummy pybossa or mturk platform, ALL platforms need to give a name, pybossa & mturk also need to give url,user,password.